File: src/algorithm/leetcode/Problem206.java
package algorithm.leetcode;

public class Problem206 {
    public static ListNode reverseList(ListNode head) {
        ListNode p = head;
        ListNode pre = null;
        ListNode aft = null;
        ListNode reverseHead = null;
        while (p != null) {
            aft = p.next;
            p.next = pre;
            if (aft == null)
                reverseHead = p;
            pre = p;
            p = aft;
        }
        return reverseHead;
    }
}

File: src/algorithm_cpp/problem206.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
	if (head == nullptr) return head;
	ListNode *node = head;
	ListNode *next = node->next;
	while (next != nullptr) {
	    ListNode *nextnext = next->next;
	    next->next = node;
	    node = next;
	    next = nextnext;
	}
	head->next = nullptr;
	return node;
    }
};
